irohad/main/application.cpp

```diff
@@ -258,36 +253,37 @@ Irohad::RunResult Irohad::initStorage(
       vm_caller_ref = *vm_caller_.value();
     }
 
-    return ::iroha::initStorage(*pg_opt_,
-                                pool_wrapper_,
-                                pending_txs_storage_,
-                                query_response_factory_,
-                                config_.block_store_path,
-                                vm_caller_ref,
-                                log_manager_->getChild("Storage"))
+    return ::iroha::initStorage(
+               *pg_opt_,
+               pool_wrapper_,
+               pending_txs_storage_,
+               query_response_factory_,
+               config_.block_store_path,
+               vm_caller_ref,
+               [this](std::shared_ptr<shared_model::interface::Block const>
+                          block) {
+                 iroha::getSubscription()->notify(EventTypes::kOnBlock, block);
+                 if (ordering_init and tx_processor and pending_txs_storage_
+                     and mst_storage) {
```

Comment:
It seems we can transfer ```ordering_init```, ```tx_processor```, etc via ```wptr``` argument in lambda-context and not to use ```this```.

Reply:
Not possible because capture is done before the fields are initialized.